db/migrate/20240119122623_add_id_uniqueness_constraint_to_friendships.rb

```diff
@@ -0,0 +1,5 @@
+class AddIdUniquenessConstraintToFriendships < ActiveRecord::Migration[7.1]
+  def change
+    add_check_constraint :friendships, 'sender_id <> receiver_id', name: 'not_friending_self'
+  end
+end
```

Comment:
This should be a validation in the rails model as well actually, it gives better errors. I'll add that in A While ™️